File: programmers/python 3/12973.py
# 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/12973

def solution(s):
    if len(s) % 2:
        return 0

    stack = []
    for ch in s:
        if stack and stack[-1] == ch:
            stack.pop()
        else:
            stack.append(ch)

    return 1 if not stack else 0

File: programmers/swift/12973.swift
// 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/12973

func solution(_ s: String) -> Int {
    guard s.count % 2 == 0 else { return 0 }

    var stack = [Character]()
    for character in s {
        if let top = stack.last, top == character {
            stack.removeLast()
        } else {
            stack.append(character)
        }
    }

    return stack.isEmpty ? 1 : 0
}